File: kotlin/src/main/kotlin/behavioral/templateMethod/MainTemplateMethod.kt
package behavioral.templateMethod

internal abstract class Conversion {
    fun toCM(inch: Int): Float {
        return inch * 2.54f
    }

    abstract fun addCmAndInch(cm: Int, inch: Int): Float
}

internal class AddingImpl : Conversion() {
    override fun addCmAndInch(cm: Int, inch: Int): Float {
        return cm + toCM(inch)
    }
}

fun main(args: Array<String>) {
    val addingImpl = AddingImpl()
    val cm = 1
    val inch = 1
    println(cm.toString() + " cm + " + inch + " inch = " + addingImpl.addCmAndInch(cm, inch) + " cm")
}

File: python/behavioral/templateMethod/MainTemplateMethod.py
from abc import ABC, abstractmethod

class Conversion(ABC):
    def _toCM(self, inch: int) -> float:
        return float(inch) * 2.54
    @abstractmethod
    def addCmAndInch(self, cm: int, inch: int) -> float:
        pass

class AddingImpl(Conversion):
    def addCmAndInch(self, cm: int, inch: int) -> float:
        return cm + self._toCM(inch)

def main():
    addingImpl = AddingImpl()
    cm = 1
    inch = 1
    print(str(cm) + " cm + " + str(inch) + " inch = " + str(addingImpl.addCmAndInch(cm, inch)) + " cm")

main()